app/controllers/subscriptions_controller.rb

```diff
@@ -36,14 +50,18 @@ def smoke_test_address?
 
   def subscriber
     @subscriber ||= begin
-                      found = Subscriber.where("LOWER(address) = ?", address.downcase).first
+                      found = find_subscriber
                       found || Subscriber.create!(
                         address: address,
                         signon_user_uid: current_user.uid,
                       )
                     end
   end
 
+  def find_subscriber
+    @find_subscriber ||= Subscriber.where("LOWER(address) = ?", address.downcase).first
```

Comment:
Probably not related to this PR, but I wonder if we should refactor this out into a class method of `Subscriber` as I could see it being useful for looking for subscribers in the console too.

Reply:
Yes good idea

Reply:
Also if this was `Subscriber.find_by!("LOWER(address) = ?", address.downcase)` we'd not need to handle the nil case as Rails would raise the RecordNotFound exception that would automatically be converted to a 404.

Reply:
Oh and you do have find_by_address and find_by_address! already which could be overridden.

Reply:
I looked at using `find_by!` and then catching the exception in `subscriber` to create a new subscriber, but this then stops us from sending a 422 for an invalid email address for some reason. I'm going to defer changing this to a separate PR.

Reply:
Yeah I'd just go with two separate calls. Rather than having the same shared method can just have Subscriber.find_by_address(address) for the find/create and the bang variant for when you're handling a 404.